language/move-stdlib/sources/Capability.move

```diff
@@ -100,6 +112,19 @@ module Std::Capability {
     /// can succeed with this operation. A caller must pass a witness that they own the `Feature` type
     /// parameter.
     public fun acquire<Feature>(requester: &signer, _feature_witness: &Feature): Cap<Feature>
+    acquires CapState, CapDelegateState {
+        Cap<Feature>{root: validate_acquire<Feature>(requester)}
+    }
+
+    /// Acquires a linear capability token. It is up to the module which owns `Feature` to decide
```

Comment:
How does the module do this? I would expect it to require (e.g.) a boolean field in `CapState`.

Reply:
We have discussed offline a scheme how to do this, basically by establishing a design pattern where the initialize and acquire functions in the basic Capability module would always be wrapped anyway in the module which defines the `Feature` resource type. This is necessary to prevent 3rd party to highjack a capability. Each of the function in this module takes a `&Feature` witness, so whoever has control over `Feature`, governs what additional conditions have to be met for acquire, whether linear or copyable caps are exposed, etc. Check out the example for illustration.